source/main/windows/windowBounds.js

```diff
@@ -0,0 +1,121 @@
+// @flow
+import { debounce } from 'lodash-es/function';
+import {
+  STORAGE_KEYS,
+  STORAGE_TYPES,
+} from '../../common/config/electron-store.config';
+import type { StoreMessage } from '../../common/types/electron-store.types';
+
+interface Rectangle {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+interface Window {
+  getBounds(): Rectangle;
+  on(event: string, handler: Function): void;
+}
+
+interface Display {
+  workArea: Rectangle;
+}
+
+interface Screen {
+  getDisplayMatching(rect: Rectangle): Display;
+  workArea: Rectangle;
+}
+
+type SendStoreRequest = (request: StoreMessage) => mixed;
+
+function maybeRectangle(data: any): ?Rectangle {
+  return data != null &&
+    typeof data.x === 'number' &&
```

Comment:
I hope we can get rid of 'typeof checking' being more strict with the types when we move to TS

Reply:
Yeah type guards will be easier to implement in Typescript 👍